File: python-engine/market/repository/instrument_repository.py
from sqlalchemy import text

class InstrumentRepository:
    def __init__(self, session):
        self.session=session

    def get_by_symbol(self,symbol:str):
        sql=text("SELECT * FROM instruments WHERE symbol=:symbol LIMIT 1")
        return self.session.execute(sql,{"symbol":symbol}).mappings().first()

    def active(self):
        sql=text("SELECT * FROM instruments WHERE is_active=true")
        return self.session.execute(sql).mappings().all()

File: laravel-beschaedigt/app/Repositories/InstrumentRepository.php
<?php
namespace App\Repositories;
use App\Models\Instrument;
class InstrumentRepository{
public function findBySymbol(string $symbol):?Instrument{
return Instrument::where('symbol',$symbol)->first();
}
public function active(){
return Instrument::where('is_active',true);
}
}
